rocAL/rocAL_pybind/amd/rocal/fn.py

```diff
@@ -255,339 +513,578 @@ def resize(*inputs, bytes_per_sample_hint=0, image_type=0, interp_type=1, mag_fi
+    
+    output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
 
+    output_dtype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
     """
+    saturation = b.createFloatParameter(saturation) if isinstance(saturation, float) else saturation
     # pybind call arguments
-    saturation = b.CreateFloatParameter(saturation) if isinstance(saturation, float) else saturation
-    kwargs_pybind = {"input_image0": inputs[0],
-                     "is_output": False, "sat": saturation}
-    saturated_image = b.Saturation(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    kwargs_pybind = {"input_image": inputs[0],
+                     "is_output": False, "sat": saturation, "output_layout": output_layout, "output_dtype": output_dtype}
+    saturated_image = b.saturation(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (saturated_image)
 
-def ssd_random_crop(*inputs, bytes_per_sample_hint=0, num_attempts=1.0, preserve=False, seed= -1, device = None):
+
+def ssd_random_crop(*inputs, p_threshold=None, crop_area_factor=None, crop_aspect_ratio=None,
+                    crop_pos_x=None, crop_pos_y=None, num_attempts=1, device=None,
+                    all_boxes_above_threshold=True, allow_no_crop=True, ltrb=True, output_layout=types.NHWC, output_dtype=types.UINT8):
     """
-    bytes_per_sample_hint (int, optional, default = 0) – Output size hint (bytes), per sample. The memory will be preallocated if it uses GPU or page-locked memory
-    num_attempts (int, optional, default = 1) – Number of attempts.
-    preserve (bool, optional, default = False) – Do not remove the Op from the graph even if its outputs are unused.
-    seed (int, optional, default = -1) – Random seed (If not provided it will be populated based on the global seed of the pipeline)
+    inputs - the input image passed to the augmentation
+
+    p_threshold (float, optional, default = None) - threshold value used for selecting bboxes during crop generation
+
+    crop_area_factor (float, optional, default = None) - area factor used for crop generation
+    
+    crop_aspect_ratio (float, optional, default = None) - aspect ratio of the cropping windows
+
+    crop_pox_x (float, optional, default = None) - crop_x position used for crop generation
+
+    crop_pox_y (float, optional, default = None) - crop_y position used for crop generation
+
+    num_attempts (int, optional, default = 1) - number of attempts to get a crop window that matches the area factor and aspect ratio conditions
+
+    device (string, optional, default = None) - Backend used for the augmentation - either 'cpu' or 'gpu'
+    
+    output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
+
+    output_dtype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
     """
-    if(num_attempts == 1):
+    if (num_attempts == 1):
```

Comment:
Is this correct?

Reply:
I was referring to assigning _num_attempts = 20 if it is set 1?

Reply:
DALI has default value of 1 for this param so we added the same but it is set as 20 if the default value is used for SSD random crop.